### PR TITLE
feat(nibijs): mainnet custom chain function

### LINKED ISSUE #261: Fix CustomChain code to display mainnet
cataclysm-1 forgoes the `nibiru` prefix

### PR DESCRIPTION
**Pull request by Unique-Divine**
- Closes #261

## Summary by CodeRabbit

- **Documentation**
  - Added a new "Mainnet" module description and usage example in the documentation.
  - Updated various documentation links to reflect the latest source code references and line numbers.
  - Modified documentation to improve clarity and accuracy of the information provided to users.

- **New Features**
  - Introduced the "Mainnet" configuration option for easier setup and connection to the Nibiru mainnet.

- **Refactor**
  - Adjusted internal references and updated documentation to align with recent codebase changes.

### CODE REVIEW

File: packages/nibijs/docs/classes/CustomChain.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ based on its chain ID.
 **`Example`**
 
 ```ts
-export const TEST_CHAIN = new CustomChain({
+export const chain = new CustomChain({
   prefix: "nibiru",
   shortName: "testnet",
   number: 1,
-}) // v0.19.2
+}) // v0.21.43
```
Comment (cgilbe27) [inline packages/nibijs/docs/classes/CustomChain.md]: Can we just remove this?

File: packages/nibijs/src/chain/chain.ts [hunk 2]
```diff
@@ -37,11 +46,11 @@ export interface ChainIdParts {
  *
  * @example
  * ```ts
- * export const TEST_CHAIN = new CustomChain({
+ * export const chain = new CustomChain({
  *   prefix: "nibiru",
  *   shortName: "testnet",
  *   number: 1,
- * }) // v0.19.2
+ * }) // v0.21.43
```
Comment (cgilbe27) [inline packages/nibijs/src/chain/chain.ts]: Remove?

File: packages/nibijs/src/test/chain.test.ts [hunk 3]
```diff
@@ -56,13 +57,9 @@ describe("chain/chain", () => {
...
     const shortName = "cataclysm"
     const number = 1
-    const result = new CustomChain({
-      shortName,
-      number,
-      mainnet: true,
-    })
     expect(result.chainId).toEqual(`${shortName}-${number}`)
     expect(result.chainName).toEqual(`${shortName}-${number}`)
     expect(result.endptGrpc).toEqual(`grpc.nibiru.fi`)
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply packages/nibijs/src/test/chain.test.ts:73]: It's correct as grpc.nibiru.fi

File: packages/nibijs/src/chain/chain.ts [hunk 4]
```diff
@@ -136,6 +145,17 @@ export const Devnet = (chainNumber: number) =>
     number: chainNumber,
   })
 
+/** Mainnet: "Chain" configuration for the Nibiru "mainnet".
+ * ❗ Mainnet uses real funds. For more info, see
+ * <ahref="https://nibiru.fi/docs/dev/networks/">Nibiru Netwokrs</a>.
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply packages/nibijs/src/chain/chain.ts]: Fixed by e4d449a6009827dbb999b414a40c9b269948b43e

File: packages/nibijs/docs/classes/CustomChain.md (hunk 1 above)
Comment (Unique-Divine) [reply packages/nibijs/docs/classes/CustomChain.md]: Done e4d449a6009827dbb999b414a40c9b269948b43e

File: packages/nibijs/src/chain/chain.ts (hunk 2 above)
Comment (Unique-Divine) [reply packages/nibijs/src/chain/chain.ts]: Done e4d449a6009827dbb999b414a40c9b269948b43e

File: packages/nibijs/src/chain/chain.ts [hunk 5]
```diff
@@ -136,12 +145,23 @@ export const Devnet = (chainNumber: number) =>
     number: chainNumber,
   })
 
+/** Mainnet: "Chain" configuration for the Nibiru "mainnet".
+ * ❗ Mainnet uses real funds. For more info, see
+ * <a href="https://nibiru.fi/docs/dev/networks/">Nibiru Netwokrs</a>.
+ * */
+export const Mainnet = (chainNumber: number = 1) =>
```
Comment (CalicoNino) [inline packages/nibijs/src/chain/chain.ts:162]: Unrelated to this change but should we also add the heart monitor & the web socket endpoint as apart of the chain interface?

We currently create those urls in the web app

**Review by CalicoNino — APPROVED** [review]

File: packages/nibijs/src/chain/chain.ts (hunk 5 above)
Comment (CalicoNino) [reply packages/nibijs/src/chain/chain.ts:162]: https://github.com/NibiruChain/ts-sdk/pull/269

**Comment by CalicoNino** [comment]
queryBatchHandler.test.ts is failing bc testnet doesn't seem to have communityPools or delegations

**Comment by cgilbe27** [comment]
> queryBatchHandler.test.ts is failing bc testnet doesn't seem to have communityPools or delegations

Tests should be pointed at devnet-2

**Pull request merged**